engine/src/main/java/com/datatorrent/stram/stream/InlineStream.java

```diff
@@ -80,14 +88,29 @@ public void teardown()
   public void put(Object tuple)
   {
     try {
-      super.put(tuple);
+      reservoir.put(tuple);
+      if (!(tuple instanceof Tuple)) {
```

Comment:
I saw that :-) But since we are benchmarking plain stream performance and spending time and effort enhancing the same, we should probably also quantify this. Have you measured before/after?